### PR TITLE
Automate deploying updates to the macs

### PR DESCRIPTION
**Pull request by grahamc**
- Upgrade to Nix 2 across the board
 - Set up more stable garbage collection
 - Fix mac9
 - Use LnL's nix-darwin to manage the version
   of Nix deployed

To update and deploy:

    $ ./update-channels
    $ ./deploy.sh

You MUST already have setup your SSH config to know
how to connect to mac{1..9} but I'm not going to document that
here.

### CODE REVIEW

File: macs/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+for macId in `seq 1 9`; do
+    host="mac$macId"
+    SSHOPTS="$host"
+
+    if ! ssh $SSHOPTS test -f "$drv"; then
+        echo "Copying derivation to $host"
+
+        # A lame version of nix-copy-closure since we can't SSH directly in
+        # as a trusted user
```
Comment (edolstra) [inline macs/deploy.sh]: Why can't we SSH in as a trusted user?

File: macs/nix/nix-darwin.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (edolstra) [inline macs/nix/nix-darwin.json]: What is this JSON stuff for?

File: macs/deploy.sh (hunk 1 above)
Comment (grahamc) [reply macs/deploy.sh]: We could! Taking this to PM.

File: macs/nix/nix-darwin.json (hunk 2 above)
Comment (grahamc) [reply macs/nix/nix-darwin.json]: To track and pin the versions of the "channels" being deployed, to keep it from being a floating version.

File: macs/deploy.sh (hunk 1 above)
Comment (grahamc) [reply macs/deploy.sh]: Fixed!

**Comment by zimbatm** [comment]
Is this still relevant @grahamc ?

**Comment by grahamc** [comment]
Nope :)

**Pull request closed without merging**